### DISCUSSION TITLE
Feature Request: UI: Change "default" font family

### DISCUSSION THREAD

**Discussion by henry2man** [flyerhq/flutter_chat_ui#90] [Ideas & Requests] [answered]
## Is your feature request related to a problem?
Yes, I want a simple way to change the default font family "Avenir". Currently this task involves the extension of a lot of text styles, even using DefaultChatTheme.

## What solution would you suggest?
The ideal solution could be make dynamic `DefaultChatTheme`, allowing a new parameter called `defaultFontFamily` (_original, isn't it?_). Maybe we could configure some sort of a `defaultTextStyle` that is used later by the specific styles using `defaultTextStyle.copyWith()`.

## Is there any additional solution to that?
I've been trying to do my own implementation in a fork but I cannot see any "one-shot" solution. Maybe a full rewrite of `chat_theme.dart` will be required.  

## Extras
**Screenshots or videos 📸**
N/A

**Code snippets 📝**

Actual code needed for change all fonts:

```dart
DefaultFontFamily def = DefaultFontFamily();

String customFontFamily = 'Arial';

DefaultFontFamily(
...
dateDividerTextStyle: def.dateDividerTextStyle.copyWith(fontFamily:customFontFamily),
emptyChatPlaceholderTextStyle: def.emptyChatPlaceholderTextStyle.copyWith(fontFamily:customFontFamily),
inputTextStyle: def.inputTextStyle.copyWith(fontFamily:customFontFamily),
receivedMessageBodyTextStyle: def.receivedMessageBodyTextStyle.copyWith(fontFamily:customFontFamily),
receivedMessageCaptionTextStyle: def.receivedMessageCaptionTextStyle.copyWith(fontFamily:customFontFamily),
receivedMessageLinkDescriptionTextStyle: def.receivedMessageLinkDescriptionTextStyle.copyWith(fontFamily:customFontFamily),
receivedMessageLinkTitleTextStyle: def.receivedMessageLinkTitleTextStyle.copyWith(fontFamily:customFontFamily),
sentMessageBodyTextStyle: def.sentMessageBodyTextStyle.copyWith(fontFamily:customFontFamily),
sentMessageCaptionTextStyle: def.sentMessageCaptionTextStyle.copyWith(fontFamily:customFontFamily),
sentMessageLinkDescriptionTextStyle: def.sentMessageLinkDescriptionTextStyle.copyWith(fontFamily:customFontFamily),
sentMessageLinkTitleTextStyle: def.sentMessageLinkTitleTextStyle.copyWith(fontFamily:customFontFamily),
userAvatarTextStyle: def.userAvatarTextStyle.copyWith(fontFamily:customFontFamily),
userNameTextStyle: def.userNameTextStyle.copyWith(fontFamily:customFontFamily),
...
);
```

## Related issues/PRs
N/A

**Comment by demchenkoalex** [comment, marked as answer]
done with v2 https://github.com/flyerhq/flutter_chat_ui/issues/646
